src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -0,0 +1,136 @@
+
+    private fun createService (ecs_cluster_name: String, user_name: String, ecsClient: EcsClient, targetGroupArn: String) {
+
+        val alb: LoadBalancer = LoadBalancer.builder().targetGroupArn(targetGroupArn).containerPort(443).build()
+
+        val serviceBuilder = CreateServiceRequest.builder().cluster(ecs_cluster_name).loadBalancers(alb).serviceName("${user_name}_test").taskDefinition("mhf_sample_task").loadBalancers(alb).desiredCount(1).build()
+        println("Creating Service...")
+
+        try {
+            val service = ecsClient.createService(serviceBuilder)
+            println("service.responseMetadata = ${service.responseMetadata()}")
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+    }
+
+    fun getPriorityVal (rulesResponse : DescribeRulesResponse) : Int {
+
+        var consecVals : Int = 1
+        for (i in rulesResponse.rules()) {
+            if (i.priority() != "default") {
+                var intVal = Integer.parseInt(i.priority())
+                if (intVal == consecVals) consecVals = consecVals + 1
+            }
+        }
+        if (consecVals >= 1000) throw Exception("The upper limit of 1000 rules has been reached for this load balancer.")
+        return consecVals;
+    }
+
+    fun taskDefinitionWithOverride(ecs_cluster_name: String, emr_cluster_host_name: String, albName :String, user_name: String, jupyterCpu : Int=512, jupyterMemory: Int = 512) {
+
+        val credentials: AwsCredentialsProvider = credentialsService.getDefaultCredentialsProvider()
+
+        val ecsClient: EcsClient = EcsClient.builder().credentialsProvider(credentials).region(awsRegion).build()
+        val albClient: ElasticLoadBalancingV2Client = ElasticLoadBalancingV2Client.builder().credentialsProvider(credentials).region(awsRegion).build()
+
+        println("Getting alb information...")
+
+        val albRequest = DescribeLoadBalancersRequest.builder().names("orchestration-service-lb").build()
+        val albResponse: DescribeLoadBalancersResponse = albClient.describeLoadBalancers(albRequest)
+
+        println("Getting Listener information...")
+
+        val albListenerRequest = DescribeListenersRequest.builder().loadBalancerArn(albResponse.loadBalancers()[0].loadBalancerArn()).build()
+        val albListenerResponse = albClient.describeListeners(albListenerRequest)
+
+        println("Creating target group...")
+
+        val tgRequest = CreateTargetGroupRequest.builder().name("$user_name-target-group").protocol("HTTPS").vpcId(albResponse.loadBalancers()[0].vpcId()).port(443).build()
+        val targetGroupResponse = albClient.createTargetGroup(tgRequest)
+
+        println("Getting target group arn...")
+
+        val albTargetGroupRequest = albClient.describeTargetGroups(DescribeTargetGroupsRequest.builder().names("$user_name-target-group").build())
+        val albTargetGroupArn = albTargetGroupRequest.targetGroups()[0].targetGroupArn()
```

Comment:
Shouldn't be hardcoded value here. Also can we guarantee that the first returned value is the one that we want?

Reply:
it's not hardcoded, we pass in the username to make a user specific target group in AWS

Reply:
Ah yes, my bad! I missed that in the above request